### ISSUE TITLE
Interpolation in macros

### ISSUE THREAD

**Issue by wycats** [castwide/solargraph#409]
I'm trying to write a macro that converts a symbol parameter into a question-mark suffixed version of the parameter:

```ruby
class << self
# @!macro [attach] variant
#   @!attribute [r] $1?
#     @return [Boolean]
def variant(symbol)
  define_method(:"#{symbol}"?) do
    # ...
  end 
end
```

When I use this method and generate docs using YARD, it generates the appropriate methods. But it seems that solargraph doesn't pick them up.

Reading other issues, it seems that interpolation is, in general, supported in macros. Are they unsupported in this position, or am I doing something wrong?

**Comment by castwide** [comment]
Support for the `@!macro` directive is in progress. There are still some issues with accuracy and performance that need to be resolved.

**Comment by wycats** [comment]
@castwide does that mean that you intend for this syntax to work, but it's not yet implemented?

**Comment by castwide** [comment]
@wycats Yes. It's on the roadmap, but unfortunately I don't have an ETA yet.

**Comment by wycats** [comment]
@castwide I'd love to help (and I think @wagenet, a coworker of mine, would also be happy to help). We use Solargraph in a Rails app, and we're starting to add in YARD annotations to improve Solargraph's behavior. Now that we've started doing that, it's clear that this is the missing piece that would unlock a lot of Rails. That makes us really excited about Solargraph's potential, and we'd be happy to help.

If you can think of any way that we could be helpful, please let us know 😄

**Comment by castwide** [comment]
Thanks. Any help you can provide is much appreciated.

I think the simplest place to start is a method that accepts 1) a method pin with a macro, and 2) a source map. The method can then generate method pins based on the source map's calls to the macro's original pin. The inner logic is going to be hairy, but I might be able to put together some proof-of-concept code this week. After that, I suspect the bigger challenge will be to make it performant enough to use in production.

**Comment by gmartsenkov** [comment]
Has there been any development on this?
